libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -51,10 +57,21 @@ export class BookSearchComponent {
   }
 
   searchBooks() {
-    if (this.searchForm.value.term) {
-      this.store.dispatch(searchBooks({ term: this.searchTerm }));
-    } else {
-      this.store.dispatch(clearSearch());
-    }
+    // get search value from form
+    this.searchForm.valueChanges.pipe(
```

Comment:
I don't like the name if this function anymore - it's not actually searching books, it's setting up the valueChanges subscription. Why not just put all this directly in the `ngOnInit`?

Reply:
Moved valueChanges to construction, and searchBooks method call now happens in valueChange subscription